packages/template-retail-react-app/app/utils/site-utils.js

```diff
@@ -222,3 +220,20 @@ export const resolveLocaleFromUrl = (url, origin) => {
         (locale) => locale.alias === defaultLocale || locale.id === defaultLocale
     )
 }
+
+/**
+ * Extract pathname and search params from a given url
+ * @private
+ * @param url
+ * @returns {{search: (string|string), pathname: string}}
+ */
+function getPathnameAndSearch(url) {
+    // Clean up the URL by replacing double slashes with a single slash
+    const [baseUrl, searchString] = url.split('?')
+    const pathname = baseUrl.includes('://') ? baseUrl.split('/').slice(3).join('/') : baseUrl
+
+    return {
+        pathname: `${pathname.startsWith('/') ? '' : '/'}${pathname}`,
+        search: searchString ? `?${searchString}` : ''
+    }
+}
```

Comment:
I think rather than creating our own function to parse the given path/url, it's better to reuse `URL`. For example, we can do something like this:

```suggestion
function getPathnameAndSearch(url) {
    const {pathname, search} = new URL(url, 'https://www.some-domain.com')
    return {pathname, search}
}
```